python/source/io.rst

```diff
@@ -447,6 +450,27 @@ or :meth:`pyarrow.dataset.Dataset.to_batches` like you would for a local one.
     It is possible to load partitioned data also in the ipc arrow
     format or in feather format.
 
+.. warning::
+
+    If the above code throws an error most likely the reason is your
+    AWS credentials are not set. Follow these instructions to get
```

Comment:
Based on https://github.com/apache/arrow-cookbook/issues/56, it seems you can also run into errors if you *have* set AWS credentials? (in which case adding anonymous=True can help, so it might still be worth mentioning it here in the warning?)

Reply:
This would need to revert to adding `filesystem` parameter rather than a `anonymous` flag directly in the `dataset` object. Unless I'm doing something wrong.

Reply:
Yeah, I wouldn't suggest changing the actual example again to use `filesystem` instead of a URI. But in the warning box it might still be worth mentioning the `anonymous` flag (and that you can then pass it with `filesystem=`), as based on my reading of #56, it might solve similar errors. 

Now, it's not really clear to me when what is a exactly giving problems. In #56 it's having credentials set that makes reading the public bucket not work, except if you use anonymous flag, while the explanation added in the PR above seems to indicate that _setting_ credentials can help.

Reply:
Makes sense. I'll update the warning. 

However, looks like #56 was having the reverse of my problems: having the `credentials` file but still unable to connect to a public bucket. I wonder if the auth code on our end is buggy or the AWS throws...

Reply:
I experimented and it seems, for me at least, the S3 URL works if I have no credentials file (presumably it is anonymous in that case) and it works if I have credentials belonging to the Ursa organization in S3.  It fails if I have invalid credentials (not an interesting use case) and if I have valid credentials that aren't part of the Ursa organization (this is the case I think that people keep running into unexpectedly, which is aided by the anonymous=True flag)